Extensions/Firebase/C_firebasetools/D_cloudfirestoretools.js

```diff
@@ -0,0 +1,296 @@
+    });
+};
+
+/**
+ * Gets a field of a document and store it in a variable.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document.
+ * @param {string} field - The field to get.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operation was successful.
+ */
+gdjs.evtTools.firebase.firestore.getField = function (
+  collectionName,
+  documentName,
+  field,
+  callbackValueVariable,
+  callbackStateVariable
+) {
+  firebase
+    .firestore()
+    .collection(collectionName)
+    .doc(documentName)
+    .get()
+    .then(function (doc) {
+      if (callbackStateVariable) callbackStateVariable.setString('ok');
+      gdjs.evtTools.network._objectToVariable(
+        doc.get(field),
+        callbackValueVariable
+      );
+    })
+    .catch(function (error) {
+      if (callbackStateVariable) callbackStateVariable.setString(error.message);
+    });
+};
+
+/**
+ * Checks for existence of a document.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document to check.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operation was successful.
+ */
+gdjs.evtTools.firebase.firestore.hasDocument = function (
+  collectionName,
+  documentName,
+  callbackValueVariable,
+  callbackStateVariable
+) {
+  firebase
+    .firestore()
+    .collection(collectionName)
+    .doc(documentName)
+    .get()
+    .then(function (doc) {
+      if (callbackStateVariable) callbackStateVariable.setString('ok');
+      callbackValueVariable.setNumber(doc.exists ? 1 : 2);
+    })
+    .catch(function (error) {
+      if (callbackStateVariable) callbackStateVariable.setString(error.message);
+    });
+};
+
+/**
+ * Checks for existence of a Field.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document.
+ * @param {string} field - The field to check.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operation was successful.
+ */
+gdjs.evtTools.firebase.firestore.hasField = function (
+  collectionName,
+  documentName,
+  field,
+  callbackValueVariable,
+  callbackStateVariable
+) {
+  firebase
+    .firestore()
+    .collection(collectionName)
+    .doc(documentName)
+    .get()
+    .then(function (doc) {
+      if (callbackStateVariable) callbackStateVariable.setString('ok');
+      callbackValueVariable.setNumber(doc.get(field) === undefined ? 2 : 1);
```

Comment:
What does this 1 and 2 represents? Do you think a string would be easier to use?

Reply:
1 is true, 2 false. I don'T use 0 as it is the default value of variables.

Reply:
In most languages based on C, 0 is false and any other value is casted to true - so better either follow this convention or use "true"/"false".